labdao/plex
# web3auth login for gateway frontend

## Summary

This PR adds the Web3Auth Plug n Play modal as the login mechanism for the gateway's frontend. Users can log in via sign-in with Google/GitHub or by entering an email address and verifying by clicking the link.

Wallet address and email address both display on the frontend. Next steps are to implement the wagmi connectors and add Metamask login as an option to the modal.

## Steps to Test

1. Boot up frontend and backend
2. Click Login Button

![Screenshot 2023-10-02 at 7 34 47 AM](https://github.com/labdao/plex/assets/29086101/ee07b146-e099-41be-803a-67b978115a8a)
3. Login through any of the methods listed

![Screenshot 2023-10-02 at 7 35 25 AM](https://github.com/labdao/plex/assets/29086101/f6f9c093-5ef7-4ad5-8ac8-be9f9765be1f)
4. Verify that an entry is created to the `users` table of the DB

<img width="1096" alt="Screenshot 2023-10-02 at 7 35 55 AM" src="https://github.com/labdao/plex/assets/29086101/68930ece-962d-4c47-9d76-ad1498421e9d">

5. Confirm frontend has email address and wallet address as state

![Screenshot 2023-10-02 at 7 35 44 AM](https://github.com/labdao/plex/assets/29086101/fc214253-f998-49d3-9e1d-38151f4bdb3b)
